lib/statsd.rb

```diff
@@ -462,7 +462,11 @@ def send_to_socket(message)
       # send(2) is atomic, however, in stream cases (TCP) the socket is left
       # in an inconsistent state if a partial message is written. If that case
       # occurs, the socket is closed down and we retry on a new socket.
-      n = socket.write(message)
+      begin
+        n = socket.write(message)
+      rescue SystemCallError
```

Comment:
I think it would be better to specifically catch `Errno::ECONNRESET` here.

It would also be good to log the error as per the rescue block